src/test/java/mate/academy/dao/impl/RoleDaoImplTest.java

```diff
@@ -0,0 +1,58 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import mate.academy.dao.RoleDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.model.Role;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RoleDaoImplTest extends AbstractTest {
+    private RoleDao roleDao;
+
+    @Override
+    protected Class<?>[] entities() {
+        return new Class[]{Role.class};
+    }
+
+    @BeforeEach
+    void setUp() {
+        roleDao = new RoleDaoImpl(getSessionFactory());
+    }
+
+    @Test
+    void getRoleByName_validRoleName_ok() {
+        Role expected = new Role(Role.RoleName.USER);
```

Comment:
you can make static import of Role.RoleName.USER
then you will be able to use it with just name USER without class prefix
this will make you easily get rid of roleUser variable and use USER.name() instead
code should become cleaner